compiler/rustc_error_codes/src/error_codes/E0001.md

```diff
@@ -1,4 +1,4 @@
-#### Note: this error code is no longer emitted by the compiler.
+### Note: this error code is no longer emitted by the compiler.
```

Comment:
Are those changes still needed? They are markdown files, so you changed the titles from `h4` to `h3`.

Reply:
This is the 2nd point in @notriddle 's summary: "yes to changing the error index".

However, after thinking about it, I don't think it's needed. We control all the error markdown files, so we can just render them at the top level and ensure there are no `# ` in them. I've gone ahead and reverted this change.